go/libraries/doltcore/env/actions/docs.go

```diff
@@ -58,17 +62,17 @@ func SaveDocsFromRoot(ctx context.Context, root *doltdb.RootValue, dEnv *env.Dol
 }
 
 // SaveTrackedDocs writes the docs from the targetRoot to the filesystem. The working root is used to identify untracked docs, which are left unchanged.
-func SaveTrackedDocs(ctx context.Context, dEnv *env.DoltEnv, workRoot, targetRoot *doltdb.RootValue, localDocs env.Docs) error {
+func SaveTrackedDocs(ctx context.Context, drw env.DocsReadWriter, workRoot, targetRoot *doltdb.RootValue, localDocs env.Docs) error {
 	docDiffs, err := diff.NewDocDiffs(ctx, workRoot, nil, localDocs)
 	if err != nil {
 		return err
 	}
 
 	docs := removeUntrackedDocs(localDocs, docDiffs)
 
-	err = dEnv.UpdateFSDocsToRootDocs(ctx, targetRoot, docs)
+	err = drw.UpdateFSDocsToRootDocs(ctx, targetRoot, docs)
 	if err != nil {
-		localDocs.Save(dEnv.FS)
+		localDocs.Save(drw.GetFS())
```

Comment:
Definitely the wrong abstraction here... the docs read writer should know how to write to disk as necessary, not pass its filesystem to another object